File: src/main/java/io/github/slimefunguguproject/bump/utils/Keys.java
package io.github.slimefunguguproject.bump.utils;

import org.bukkit.NamespacedKey;

import io.github.slimefunguguproject.bump.implementation.Bump;

import lombok.experimental.UtilityClass;

/**
 * Some reusable {@link NamespacedKey Namespaced Keys}.
 *
 * @author ybw0014
 */
@UtilityClass
public final class Keys {
    public static final NamespacedKey APPRAISABLE = Bump.createKey("appraisable");
    public static final NamespacedKey APPRAISE_LEVEL = Bump.createKey("appraise_level");
    public static final NamespacedKey LAST_USED = Bump.createKey("last_used");
    public static final NamespacedKey SKY_SWORD_PROTECTED = Bump.createKey("sky_sword_protected");
    public static final NamespacedKey PROJECTILE = Bump.createKey("projectile");
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/utils/constant/Keys.kt
package io.github.slimefunguguproject.bump.utils.constant

import io.github.slimefunguguproject.bump.Bump
import org.bukkit.NamespacedKey
import java.util.Locale

object Keys {
    val APPRAISABLE = "appraisable".createKey()
    val APPRAISE_LEVEL = "appraise_level".createKey()
    val APPRAISE_VERSION = "appraise_version".createKey()
    val LAST_USED = "last_used".createKey()
    val SKY_SWORD_PROTECTED = "sky_sword_protected".createKey()
    val PROJECTILE = "projectile".createKey()

    fun String.createKey() = NamespacedKey(Bump.instance, this.lowercase(Locale.getDefault()))
}
